python/dlr/counter/system.py

```diff
@@ -0,0 +1,110 @@
+import platform
+import uuid
+import hashlib
+import logging
+import abc
+from abc import ABC
+
+from .deviceinfo import DeviceInfo
+
+
+# Interface
+class System:
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_device_info(self):
+        """Return a list of device information"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_device_uuid(self):
+        """Return DeviceInfo uuid"""
+        raise NotImplementedError
+
+
+# Wrapper class
+class ARM(System, ABC):
+    pass
+
+
+class Linux_ARM(ARM):
+    def __init__(self):
+        self._device = DeviceInfo()
+
+        try:
+            # retrieve device information
+            self._device.machine = platform.machine()
+            self._device.arch = platform.architecture()[0]
+            _uuid = ':'.join(
+                ['{:02x}'.format((uuid.getnode() >> ele) & 0xff) for ele in range(0, 8 * 6, 8)][::-1])
+            _md5uuid = hashlib.md5(_uuid.encode())
```

Comment:
same here, I'd suggest a separate hash helper for future auditing purpose.

Reply:
+ @mnuyens ,
We have moved hashlib functionality to new helper module.